### PR TITLE
docs: Remove old ANN references

### PR DESCRIPTION
**Pull request by ocervell**

### CODE REVIEW

File: .github/workflows/robotoff.yml [hunk 1]
```diff
@@ -74,9 +74,6 @@ jobs:
       - name: Code quality isort
         run: poetry run isort . --check
 
-      - name: Typing check mypy
```
Comment (kulizhsy) [inline .github/workflows/robotoff.yml:77]: We still want the mypy check, just without 'services/ann':

`poetry run mypy .`

File: .github/workflows/robotoff.yml [hunk 2]
```diff
@@ -74,9 +74,6 @@ jobs:
       - name: Code quality isort
         run: poetry run isort . --check
```
Comment (teolemon) [inline .github/workflows/robotoff.yml:76]:
```suggestion

  - name: Typing check mypy
  -  run: poetry run mypy . services/ann
```

Comment (teolemon) [reply .github/workflows/robotoff.yml:76]: @kulizhsy @ocervell here's a fix

**Review by kulizhsy — COMMENTED** [review]
Sorry for the confusing initial comment - we want to keep the mypy checks, but just on the '.'

mypy has a [recursive dir checking limitations](https://github.com/python/mypy/issues/8548) so 'services/ann'  was added explicitly to get around it.

**Pull request merged**